src/Exception/InvalidArgumentException.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace React\Promise\Exception;
+
+class InvalidArgumentException extends \InvalidArgumentException
```

Comment:
The use of this class appears a bit unclear. Do we really need this class? Does it make sense to add documentation for this class and/or mark this class a `@internal`?

Reply:
This is the (slipped in) introduction of named exception constructors (like eg. [doctrine does](https://github.com/doctrine/doctrine2/blob/0c4f9a886682441628f895bda38bdbb3c1703900/lib/Doctrine/ORM/ORMException.php)). I'm going to revert that as it is unrelated to the PR.

Reply:
Removed in bfe3e98